sound/soc/codecs/rt700-sdw.c

```diff
@@ -490,6 +492,34 @@ static int __maybe_unused rt700_dev_suspend(struct device *dev)
 	return 0;
 }
 
+static int __maybe_unused rt700_dev_system_suspend(struct device *dev)
+{
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+	struct rt700_priv *rt700 = dev_get_drvdata(dev);
+	int ret;
+
+	if (!rt700->hw_init)
+		return 0;
+
+	/*
+	 * prevent new interrupts from being handled after the
+	 * deferred work completes and before the parent disables
+	 * interrupts on the link
+	 */
+	mutex_lock(&rt700->disable_irq_lock);
+	rt700->disable_irq = true;
+	ret = sdw_update_no_pm(slave, SDW_SCP_INT1,
+			       SDW_SCP_INT1_IMPL_DEF, 0);
```

Comment:
@plbossart Shouldn't `SDW_SCP_INT1` be `SDW_SCP_INTMASK1` if you want to disable `SDW_SCP_INT1_IMPL_DEF`?

Reply:
yep, that's a bug @bardliao. This should be SDW_SCP_INTMASK1, good catch. fixed now.